src/Installer.php

```diff
@@ -24,14 +24,23 @@ public function install() {
 	}
 
 	/**
-	 * Modifies default page content.
+	 * Modifies default page content replacing it with classic shortcode block.
+	 * We check for shortcode as default because after WooCommerce 8.3, block based checkout is used by default.
+	 * This only runs on Tools > Create Pages as the filter is not applied on WooCommerce plugin activation.
 	 *
 	 * @param array $pages Default pages.
 	 * @return array
 	 */
 	public function create_pages( $pages ) {
-		$pages['cart']['content']     = '<!-- wp:woocommerce/classic-shortcode {"shortcode":"cart"} /-->';
-		$pages['checkout']['content'] = '<!-- wp:woocommerce/classic-shortcode {"shortcode":"checkout"} /-->';
+
+		if ( '<!-- wp:shortcode -->[woocommerce_cart]<!-- /wp:shortcode -->' === $pages['cart']['content'] ) {
```

Comment:
So we're only replacing if this is the full content of the page? not a single line jump or extra filter? Why not replace `'<!-- wp:shortcode -->[woocommerce_cart]<!-- /wp:shortcode -->'` with `'<!-- wp:woocommerce/classic-shortcode {"shortcode":"cart"} /-->'`?

Reply:
You are right, but to keep previous behaviour (and possible bugs but not introduce new ones) I chose to target that specific content (most probably coming from woocommerce_create_pages default filter value).

This won't actually run on WC activation due to it being done in a child process. Currently it only applies to Tools > Create pages.
It is important to ship this on 10.4.0 so that functionality doesn't create pages with shortcodes for users with blocks 10.4.0 but WC <8.3
That means WC 8.3 PR will only be fully testable by including this.

We can revisit this later, my plan was not to rock the boat too much right now.

Reply:
I agree this makes sense now that I reviewed the PR in core.